src/state/Maps/selectors.js

```diff
@@ -1,1181 +1,482 @@
- * @param layer {Object} layer state
- * @param activeKeys {Object} Metadata active keys
- * @param useMetadata {Object}
- * @return {{filter, filterByActive, mergedFilter, layer}}
- */
-function getFiltersForUse_deprecated(layer, activeKeys, useMetadata) {
-	let filter = {};
-	let filterByActive = {};
-	let mergedFilter = {};
-
-	if (layer && layer.hasOwnProperty('scope')){
-		filter.scopeKey = layer.scope;
-	} else if(!useMetadata || (useMetadata && useMetadata.scope)) {
-		filterByActive.scope = true;
-		if (activeKeys && activeKeys.activeScopeKey) {
-			mergedFilter.scopeKey = activeKeys.activeScopeKey;
-		}
-	}
-	if (layer && layer.hasOwnProperty('place')){
-		filter.placeKey = layer.place;
-	} else if(!useMetadata || (useMetadata && useMetadata.place)) {
-		filterByActive.place = true;
-		if (activeKeys && activeKeys.activePlaceKey) {
-			mergedFilter.placeKey = activeKeys.activePlaceKey;
-		}
-	}
-	if (layer && layer.hasOwnProperty('period')){
-		filter.periodKey = layer.period;
-	} else if(!useMetadata || (useMetadata && useMetadata.period)) {
-		filterByActive.period = true;
-		if (activeKeys && activeKeys.activePeriodKey) {
-			mergedFilter.periodKey = activeKeys.activePeriodKey;
-		}
-	}
-	if (layer && layer.hasOwnProperty('case')){
-		filter.caseKey = layer.case;
-	} else if(!useMetadata || (useMetadata && useMetadata.case)) {
-		filterByActive.case = true;
-		if (activeKeys && activeKeys.activeCaseKey) {
-			mergedFilter.caseKey = activeKeys.activeCaseKey;
-		}
-	}
-	if (layer && layer.hasOwnProperty('scenario')){
-		filter.scenarioKey = layer.scenario;
-	} else if(!useMetadata || (useMetadata && useMetadata.scenario)) {
-		filterByActive.scenario = true;
-		if (activeKeys && activeKeys.activeScenarioKey) {
-			mergedFilter.scenarioKey = activeKeys.activeScenarioKey;
-		}
-	}
-
-	if (layer && layer.hasOwnProperty('layerTemplate')){
-		filter.layerTemplateKey = layer.layerTemplate;
-	}
-
-	if (layer && layer.hasOwnProperty('attribute')){
-		filter.attributeKey = layer.attribute;
-	} else if(!useMetadata || (useMetadata && useMetadata.attribute)) {
-		filterByActive.attribute = true;
-		if (activeKeys && activeKeys.activeAttributeKey) {
-			mergedFilter.attributeKey = activeKeys.activeAttributeKey;
-		}
-	}
-
-	mergedFilter = {...filter, ...mergedFilter};
+const getAllLayersStateByMapKey = createCachedSelector(
+    [
+        getBackgroundLayerStateByMapKey,
+        getLayersStateByMapKey
+    ],
+    (backgroundLayer, layers) => {
+        if (layers || backgroundLayer) {
+            return helpers.mergeBackgroundLayerWithLayers(backgroundLayer, layers);
+        } else {
+            return null;
+        }
+    }
+)((state, mapKey) => mapKey);
 
-	return {
-		layer: layer ? layer : null,
-		filter,
-		filterByActive,
-		mergedFilter
-	}
-}
+// TODO add logic
+const getMapBackgroundLayer = createCachedSelector(
+    [
+        (state, mapKey) => getBackgroundLayerStateByMapKey(state, mapKey)
```

Comment:
does this need the arguments written out?

Reply:
It is work in progress here. The combiner will be probably moved to helper. See: https://github.com/gisat-panther/ptr-state/blob/feature/data-endpoint/src/state/_deprecatedMaps/selectors.js#L627